### PR TITLE
Gmail oneclick

### LINKED ISSUE #265: Make HTML version of authorisation emails
Came up in #241, I'm currently working on this in https://github.com/dstansby/camdram/tree/gmail-oneclick

### PR DESCRIPTION
**Pull request by dstansby**
This now fixes #265 - currently the HMTL is very basic, but it all works, and is very much open to reformatting as people think is best!

I've also fixed a couple of typos in a few places.

### CODE REVIEW

**Comment by dstansby** [comment]
Hmm, the tests seem to object to me adding the HTML bit to src/Acts/CamdramBundle/Service/EmailDispatcher.php

**Comment by dstansby** [comment]
I've commented on the problem in the "Files Changed" tab above

File: src/Acts/CamdramBundle/Service/EmailDispatcher.php [hunk 1]
```diff
@@ -44,16 +44,29 @@ public function sendShowCreatedEmail(Show $show, array $owners, array $moderator
...
+                    array(
+                        'owners' => $owners,
+                        'show' => $show,
+                    )
+                ), 'text/html'
+            )
+            
+            /* Plain Text */
+            ->addPart(
+                $this->twig->render(
```
Comment (dstansby) [inline src/Acts/CamdramBundle/Service/EmailDispatcher.php]: This is a second call to "render", whereas the test only expects one call (as before when there was only plain). I presume the fix is to remove the test? (I don't know if you can test for render being called twice?)

Comment (hoyes) [inline src/Acts/CamdramBundle/Service/EmailDispatcher.php]: Try changing `$this->once()` to `$this->exactly(2)` or `$this->atLeastOnce()` in https://github.com/camdram/camdram/blob/master/src/Acts/CamdramBundle/Tests/Service/EmailDispatcherTest.php#L55
Relevant bit of PHPUnit docs: https://phpunit.de/manual/current/en/test-doubles.html#test-doubles.mock-objects.tables.matchers

**Comment by dstansby** [comment]
Great, it seems to have passed now, thanks a lot for the tip! Also, apologies for the commit history mess, I'm slowly getting there with git...

**Comment by hoyes** [comment]
This looks great. I've currently got WIP in my local checkout but I'll take a proper look and merge it in later on.

To be honest I'm not really familiar with rebasing either, but looks useful for future use: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Comment by dstansby** [comment]
So I just had a go at all that rebasing stuff, and it turns out that it only really works if you do it before submitting a pull request! Thanks for mentioning it though.

**Comment by hoyes** [comment]
Ah well not to worry!

**Comment by dstansby** [comment]
For reference, I just tried doing a rebase, and although in theory a good
idea, I think it really messes with things when you're editing stuff on the
server. Seems a lot cleaner just to merge in from master every so often if
your branch skips behind.

**Pull request merged**